packages/instrumentation-mongoose/src/utils.ts

```diff
@@ -80,8 +80,8 @@ export function handlePromiseResponse(
     responseHook?: MongooseResponseCustomAttributesFunction
 ): any {
     if (!(execResponse instanceof Promise)) {
-        span.end();
         applyResponseHook(span, execResponse, responseHook);
+        span.end();
```

Comment:
Can you please add a simple test case for this fix?

Reply:
Sure thing!

I did confirm there currently is no test coverage of this path, but it does seem like a weird niche use case that mongoose responses _shouldn't_ be hitting anyway 😅 .

I tested the compiled package on our prod codebase and this fixed the issue we were having, but I'm having issues reproducing a test to hit this path, so it may take some time to figure out.

Reply:
> but it does seem like a weird niche use case that mongoose responses shouldn't be hitting anyway

You had this issue in your app, right? maybe you can get inspiration on how to write a test that will trigger this flow?